clients/geth/specular/rollup/services/sequencer/sequencer.go

```diff
@@ -62,28 +62,102 @@ func New(eth services.Backend, proofBackend proof.Backend, cfg *services.Config,
 	return s, nil
 }
 
+// Removes tx from slice
+func removeTx(slice []*types.Transaction, elem *types.Transaction) []*types.Transaction {
+	for i := len(slice) - 1; i >= 0; i-- {
+		if slice[i].Hash() == elem.Hash() {
+			slice := append(slice[:i], slice[i+1:]...)
+			return slice
+		}
+	}
+	return slice
+}
+
 // This goroutine fetches txs from txpool and batches them
-// TODO: fetching txs through sub is unstable
-// TODO: batch txs by timer and directly querying txpool as worker does
 func (s *Sequencer) batchingLoop() {
 	defer s.Wg.Done()
 	defer close(s.batchCh)
 
+	// Ticker
+	var ticker = time.NewTicker(timeInterval)
+	defer ticker.Stop()
+
 	// Watch transactions in TxPool
 	txsCh := make(chan core.NewTxsEvent, 4096)
 	txsSub := s.Eth.TxPool().SubscribeNewTxsEvent(txsCh)
 	defer txsSub.Unsubscribe()
 
+	// Process txns via batcher
 	batcher, err := NewBatcher(s.Config.Coinbase, s.Eth)
 	if err != nil {
 		log.Crit("Failed to start batcher", "err", err)
 	}
 
+	var batchTxs []*types.Transaction
+
+	// Loop over txns
 	for {
 		select {
-		case ev := <-txsCh:
-			// New transactions from TxPool
-			err = batcher.CommitTransactions(ev.Txs)
+		case <-ticker.C:
+			// Get pending txs - locals and remotes, sorted by price
+			var txs []*types.Transaction
+			var sortedTxs *types.TransactionsByPriceAndNonce
+			signer := types.MakeSigner(batcher.chainConfig, batcher.header.Number)
+
+			pending := s.Eth.TxPool().Pending(true)
+			localTxs, remoteTxs := make(map[common.Address]types.Transactions), pending
+			for _, account := range s.Eth.TxPool().Locals() {
+				if txs = remoteTxs[account]; len(txs) > 0 {
+					delete(remoteTxs, account)
+					localTxs[account] = txs
+				}
+			}
+			if len(localTxs) > 0 {
+				sortedTxs = types.NewTransactionsByPriceAndNonce(signer, localTxs, batcher.header.BaseFee)
+			}
+			if len(remoteTxs) > 0 {
+				sortedTxs = types.NewTransactionsByPriceAndNonce(signer, remoteTxs, batcher.header.BaseFee)
+			}
+
+			// Batch txns
+			if sortedTxs == nil {
+				continue
```

Comment:
we still need to send `batchTxs` if there is nothing from the pool (although seems not possible)